packages/beacon-node/src/chain/chain.ts

```diff
@@ -646,12 +647,12 @@ export class BeaconChain implements IBeaconChain {
     };
   }
 
-  recomputeForkChoiceHead(): ProtoBlock {
+  recomputeForkChoiceHead(mode: UpdateHeadOpt = UpdateHeadOpt.GetCanonicialHead, slot?: Slot): ProtoBlock {
```

Comment:
it takes me a while to figure out the use of different params here. I prefer to keep this function as is and add:
- `predictProposerHead`: to be used in `prepareNextSlot.ts`
- `getProposerHead` when we produce a block

that'd be cleaner to me, each function has its own purpose. Would like to know opinions from @wemeetagain @g11tech too

Reply:
> it takes me a while to figure out the use of different params here. I prefer to keep this function as is and add:
> 
> * `predictProposerHead`: to be used in `prepareNextSlot.ts`
> * `getProposerHead` when we produce a block
> 
> that'd be cleaner to me, each function has its own purpose. Would like to know opinions from @wemeetagain @g11tech too

Do you think we should split the `forkchoice.updateAndGetHead()` too?